.github/workflows/coverage.yml

```diff
@@ -127,24 +138,20 @@ jobs:
       - name: Run Pulsar service
         run: DOCKER_SERVICES=pulsar make docker-compose-up
 
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: 1.62
-          override: true
-          components: clippy, llvm-tools-preview, rustfmt
+      - name: Install Rust
+        run: rustup update stable
 
-      - uses: taiki-e/install-action@v1
+      - uses: taiki-e/install-action@v2
         with:
-          tool: cargo-llvm-cov,protoc
-
-      - uses: Swatinem/rust-cache@v2
+          tool: cargo-llvm-cov,nextest,protoc
 
+      # We limit the number of jobs to 4 to avoid OOM errors when linking the binary.
       - name: Generate code coverage
         run: |
           cargo llvm-cov clean --workspace
-          cargo llvm-cov --test failpoints --no-report --features fail/failpoints
-          cargo llvm-cov --no-report --all-features
-          cargo llvm-cov --no-run --lcov --output-path lcov.info
+          cargo llvm-cov nextest --no-report --test failpoints --features fail/failpoints --retries 2
+          CARGO_BUILD_JOBS=4 cargo llvm-cov nextest --no-report --all-features --retries 2
```

Comment:
Can we go a bit higher? 5 or 6?

Reply:
I think hiding flaky tests here by increasing the number might create some issues for use during normal PR runs.

Reply:
Can we start at 2, then gradually go to 0? We still have a few flaky tests that have not been fixed. Moreover, unless you commit to Azure storage, Kafka, Kinesis, and Pulsar sources, which we seldom do, the "regular" CI workflow should have already been executed on our code.

Reply:
I think 2 is reasonable at the moment, my comment was against going to 5-6. 2 gives test a second chance and works great reducing noise from low frequency failures. But if something requires 3+ attempts it to succeeded it is no longer a low frequency failure and should probably be fixed.

Reply:
I was talking about the number of compile jobs :D

Reply:
Let's stick with 4 as discussed offline.